lib/dataset/include/scipp/dataset/aligned_dict.h

```diff
@@ -4,45 +4,23 @@
 /// @author Simon Heybrock
 #pragma once
 
-#include <boost/container/small_vector.hpp>
-#include <boost/iterator/transform_iterator.hpp>
-
+#include "scipp/core/dict.h"
 #include "scipp/core/sizes.h"
 #include "scipp/core/slice.h"
-#include "scipp/dataset/map_view_forward.h"
+#include "scipp/dataset/aligned_dict_forward.h"
 #include "scipp/units/dim.h"
 #include "scipp/units/unit.h"
 #include "scipp/variable/logical.h"
 #include "scipp/variable/variable.h"
 
 namespace scipp::dataset {
-
-namespace detail {
-struct make_key_value {
-  template <class T> auto operator()(T &&view) const {
-    using View =
-        std::conditional_t<std::is_rvalue_reference_v<T>, std::decay_t<T>, T>;
-    return std::pair<std::string, View>(view.name(), std::forward<T>(view));
-  }
-};
-
-struct make_key {
-  template <class T> auto operator()(T &&view) const { return view.first; }
-};
-
-struct make_value {
-  template <class T> auto operator()(T &&view) const { return view.second; }
-};
-
-} // namespace detail
-
-template <class T>
-auto slice_map(const Sizes &sizes, const T &map, const Slice &params) {
-  T out;
+template <class Mapping>
+Mapping slice_map(const Sizes &sizes, const Mapping &map, const Slice &params) {
+  Mapping out;
   for (const auto &[key, value] : map) {
     if (value.dims().contains(params.dim())) {
       if (value.dims()[params.dim()] == sizes[params.dim()]) {
-        out[key] = value.slice(params);
+        out.insert_or_assign(key, value.slice(params));
```

Comment:
But insert silently does nothing if the key already exists. Is this really better?

Reply:
I am wondering if our C++ naming should mirror Python naming in more cases, e.g., call it `setitem`? Not a request to change here, just pondering...

Reply:
This PR tries to match a Python dict as closely as possible. So it would make sense to match its naming. Should we do a pass over several types (at least Dict, AlignedDict, Dataset) to change to Python names? As you said, this can be done later.

Reply:
I think `Dict` should maybe stick to the C++ naming, since it is meant as a (nearly) drop-in replacement for a C++ map. But everything that we bind to Python (such as instances of `AlignedDict`) would be a candidate for a rename?

Reply:
I would keep `Dict` and `SizedDict` (renamed) consistent with each other.